packages/blocks/src/lib/slam-map-2d/points.svelte

```diff
@@ -0,0 +1,154 @@
+<!--
+  @component
+  Renders THREE.Points from a .pcd file.
+  Creates an invisible plane mesh with dimensions matching the diameter of
+  the points' bounding sphere.
+  Emits click events that intersect this plane.
+-->
+<script lang='ts'>
+
+import * as THREE from 'three';
+import { PCDLoader } from 'three/examples/jsm/loaders/PCDLoader';
+import { T, useThrelte, createRawEventDispatcher, extend } from '@threlte/core';
+import { MeshDiscardMaterial, MouseRaycaster } from 'trzy';
+import { renderOrder } from './constants';
+import { onMount } from 'svelte';
+
+extend({ MeshDiscardMaterial });
+
+/** A buffer representing a .pcd file */
+export let pointcloud: Uint8Array | undefined;
+
+/** The size of each individual point */
+export let size: number;
+
+type $$Events = {
```

Comment:
Why `$$Events` here and `Events` in `packages/blocks/src/lib/slam-map-2d/index.svelte`?

Does declaring the type separately show up in IDE suggestions as expected?

Reply:
`$$Events`` has a different effect: it singles complete changes to event dispatcher payload rather than the contents of a custom event's detail object. It's used within threlte applications because of the usage of `createRawEventDispatcher`, which bypasses using CustomEvent objects since we're working in webgl rather than HTML-land. All intellisense is preserved 👍